pkg/channel/message_receiver.go

```diff
@@ -155,10 +153,6 @@ func (r *MessageReceiver) ServeHTTP(response nethttp.ResponseWriter, request *ne
 	}
 
 	transformers := append(defaultTransformers, AddHistory(host)...)
-	span := trace.FromContext(request.Context())
-	if span != nil {
-		transformers = append(transformers, tracing.AddTraceparent(span)...)
-	}
```

Comment:
According to https://github.com/cloudevents/spec/blob/master/extensions/distributed-tracing.md, the tracing extensions should still be set even though the HTTP headers will be used for trace propagation.

Reply:
To be honest this is not clear to me, because first it states that:

> the Distributed Tracing attributes MUST be encoded over HTTP(S) as headers

but then it talks about conflicts... I think this needs some clarification in the spec itself

Reply:
I agree it could use clarification. If the CE attributes are optional over HTTP we should avoid setting them. In this case maybe we should even clear them when already present in the event?

Reply:
That's an interesting question... I suppose we should, and maybe at this point it just makes sense to keep the initial `AddTraceparent`... Can you open an issue so we can find out the solution in a followup? I wanna have this one merged to unblock the #2813

Reply:
I've created #2918. In the meantime I think it's OK to drop AddTraceparent if you prefer.